### PR TITLE
feat(index js): add requiredIf to named exports

### PR DESCRIPTION
**Pull request by Mohammer5**
~~@HendrikThePendric for some reasons, when I commit changes, the pre-commit hook just emptied the README, no idea what went wrong here. So this is something we need to fix, do you have an idea what went wrong (#76)?~~

EDIT: My local deps weren't up2date

### CODE REVIEW

**Comment by HendrikThePendric** [comment]
In the pre-commit hook, the docs are built. And for this repo that means a new readme is (or rather "should be") generated based on the template `./jsdoc2md/README.hbs` and the jsDoc comments in the files in `./src`. So it seems this is failing locally for you. Could you first check what happens when you run `yarn docs`?

If it fails, I don't know if fixing it locally is a good way forward. Perhaps we should just focus on building the docs in travis and making sure it works there.

File: src/requiredIf.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ const requiredIfFactory = (condition, propType, isRequired) => (
     componentName,
     _location,
     propFullName // normally null but a string like "propName[index]" when wrapped in arrayOf
+    // eslint-disable-next-line max-params
```
Comment (HendrikThePendric) [inline src/requiredIf.js]: It should not be necessary to add that line. This repo has the following ESLint rule:
```javascript
'max-params': ['error', { max: 5 }],
```

Comment (Mohammer5) [reply src/requiredIf.js]: well, I couldn't commit anything without this one. Seems like something is going wrong on my local system...

Comment (Mohammer5) [reply src/requiredIf.js]: My local deps weren't up2date. I've removed that line

**Pull request merged**